File: 268_missing_number.java
import java.util.Arrays;

class missing_number_268 {
    public int missingNumber(int[] nums) {
        int actualSum = Arrays.stream(nums).sum();
        int expectedSum = (nums.length * (nums.length + 1))/2;
        return expectedSum - actualSum;
    }
}

File: 268_MissingNumber.py
from typing import List

# tc = O(n)
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n=len(nums)
        missing = 0
        for i in range(n):
            missing=missing^nums[i]^(i+1)
        
        return missing
    

# # tc = O(n)
# class Solution:
#     def missingNumber(self, nums: List[int]) -> int:
#         n = len(nums)
#         return ((n*(n+1))//2) - sum(nums)